book/tutorials/basic-git.md

```diff
@@ -0,0 +1,201 @@
+# Basic Git/GitHub Skills Using `jupyterlab-git`
+
+```{admonition} Learning Objectives
+- Set up Git in the JupyterHub using [First time `git` setup](../preliminary/git)
+- Learn how to use the `jupyterlab-git` GUI in JupyterLab
+- Practice 4 basic Git/GitHub skills: cloning, committing, push/pull
+```
+
+```{important}
+Before we go over this tutorial we expect that you have already gone through this preparation material:
+- [JupyterHub login setup](../preliminary/jupyterhub)
+```
+
+## What is Git and GitHub?
+
+**Git** A program to track your file changes and create a history of those changes. Creates a 'container' for a set of files called a repository.
+
+**GitHub** A website to host these repositories and allow you to sync local copies (on your computer) to the website. *Lots* of functionality built on top of this.
+
+![](../img/git-repo-structure.png){width=200px}
+
+## Some basic Git jargon
+
+* **Repo** Repository. It is your code and the record of your changes. This record and also the status of your repo is a hidden folder called `.git` . You have a local repo and a remote repo. The remote repo is on GitHub (for in our case) is called `origin`. The local repo is on the JupyterHub.
+* **Stage** Tell Git which changes you want to commit (write to the repo history).
+* **Commit** Write a note about what change the staged files and "commit" that note to the repository record. You are also tagging this state of the repo and you could go back to this state if you wanted.
+* **Push** Push local changes (commits) up to the remote repository on GitHub (`origin`).
+* **Pull** Pull changes on GitHub into the local repository on the JupyterHub.
+* **Git GUIs** A graphical interface for Git (which is command line). Today I will use `jupyterlab-git` which we have installed on JupyterHub.
+* **Shell** A terminal window where we can issue `git` commands.
+
+
+## Overview
+
+Today I will cover the four basic Git/GitHub skills. We will not work with branches today and I won't cover much about merge conflicts. The goal for today is to first get you comfortable with the basic skills and terminology---and also get you set up on the JupyterHub. We will use what is called a  "trunk-based workflow".
+
+### Simple Trunk-based Workflow:
+
+* Make local (on your computer) changes to code.
+* Record what those changes were about and commit to the code change record (history).
+* Push those changes to your remote repository (aka origin)
+
+We'll do this
+
+![](../img/git-linear-flow-2.png)
+
+## Setting up Git
+
+Before we can work with Git in JupyterLab, we need to do some set up.
+
+[First time `git` setup](../preliminary/git)
+
+1. Tell Git who you are and to store your credentials (GitHub login info)
+
+[Show me](https://youtu.be/3CLuOCJMfK0)
```

Comment:
It looks like you can use the first frame of the video as a thumbnail if you want: 

[![GitHub Login Video](http://img.youtube.com/vi/3CLuOCJMfK0/0.jpg)](http://www.youtube.com/watch?v=3CLuOCJMfK0 "GitHub Login Video")

https://stackoverflow.com/questions/11804820/how-can-i-embed-a-youtube-video-on-github-wiki-pages

Reply:
Actually based on [this comment](https://github.com/executablebooks/jupyter-book/issues/1249#issuecomment-924221631) I think you can embed the video with player controls so that people don't have a new tab to youtube open up: 

> <iframe width="560" height="315" src="https://www.youtube.com/embed/3CLuOCJMfK0" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>